src/test/java/catascore/KeyGeneratorImplTest.java

```diff
@@ -10,21 +10,45 @@
     final static int N_CHAR = 6;
 
     @Test
-    public void generateSixCharacterKey() {
+    public void generateSixCharacterKey() throws Exception{
         KeyGenerator keyGenerator = new KeyGeneratorImpl();
         assertEquals(N_CHAR, keyGenerator.generateKey().length());
     }
 
     @Test
-    public void generateUppercaseKey() {
+    public void generateUppercaseKey() throws Exception{
         KeyGenerator keyGenerator = new KeyGeneratorImpl();
-        String key = keyGenerator.generateKey();
+        String key = null;
+        key = keyGenerator.generateKey();
         assertEquals(key.toUpperCase(),key);
     }
 
     @Test
-    public void  generateUniqueKey() {
+    public void  generateUniqueKey() throws Exception{
         KeyGenerator keyGenerator = new KeyGeneratorImpl();
         assertNotEquals(keyGenerator.generateKey(),keyGenerator.generateKey());
     }
+
+    @Test
+    public void changeNextLetter() throws Exception{
+        KeyGenerator keyGenerator = new KeyGeneratorImpl();
+            String key;
+        do{
+                key = keyGenerator.generateKey();
+        }while (key.charAt(key.length() -1) != 'Z');
+
+        key = keyGenerator.generateKey();
+        assertEquals('A', key.charAt(key.length() -1));
+        assertEquals('B', key.charAt(key.length() -2));
+    }
```

Comment:
I think we could make a private method that receives two indexes, one from the beginning and one from the end to complete it with Z and then we could try the last case.

Reply:
Is not possible